packages/ngtools/webpack/src/transformers/ctor-parameters.ts

```diff
@@ -202,49 +202,29 @@ export function decoratorDownlevelTransformer(
   diagnostics: ts.Diagnostic[],
 ): (context: ts.TransformationContext) => ts.Transformer<ts.SourceFile> {
   return (context: ts.TransformationContext) => {
-    /** A map from symbols to the identifier of an import, reset per SourceFile. */
-    let importNamesBySymbol = new Map<ts.Symbol, ts.Identifier>();
+    const parameterTypeSymbols = new Set<ts.Symbol>();
 
     /**
      * Converts an EntityName (from a type annotation) to an expression (accessing a value).
      *
      * For a given ts.EntityName, this walks depth first to find the leftmost ts.Identifier, then
      * converts the path into property accesses.
      *
-     * This generally works, but TypeScript's emit pipeline does not serialize identifiers that are
-     * only used in a type location (such as identifiers in a TypeNode), even if the identifier
-     * itself points to a value (e.g. a class). To avoid that problem, this method finds the symbol
-     * representing the identifier (using typeChecker), then looks up where it was imported (using
-     * importNamesBySymbol), and then uses the imported name instead of the identifier from the type
-     * expression, if any. Otherwise it'll use the identifier unchanged. This makes sure the
-     * identifier is not marked as stemming from a "type only" expression, causing it to be emitted
-     * and causing the import to be retained.
      */
     function entityNameToExpression(name: ts.EntityName): ts.Expression | undefined {
-      const sym = typeChecker.getSymbolAtLocation(name);
-      if (!sym) {
-        return undefined;
-      }
-      // Check if the entity name references a symbol that is an actual value. If it is not, it
-      // cannot be referenced by an expression, so return undefined.
-      let symToCheck = sym;
-      if (symToCheck.flags & ts.SymbolFlags.Alias) {
-        symToCheck = typeChecker.getAliasedSymbol(symToCheck);
-      }
-      if (!(symToCheck.flags & ts.SymbolFlags.Value)) {
-        return undefined;
-      }
-
       if (ts.isIdentifier(name)) {
-        // If there's a known import name for this symbol, use it so that the import will be
-        // retained and the value can be referenced.
-        const value = importNamesBySymbol.get(sym);
-        if (value) {
-          return value;
+        const typeSymbol = typeChecker.getSymbolAtLocation(name);
+        if (typeSymbol) {
+          parameterTypeSymbols.add(typeSymbol);
         }
 
-        // Otherwise this will be a locally declared name, just return that.
-        return name;
+        // Based on TS's strategy to allow the checker to reach this identifier
```

Comment:
Do you think we might need to change this if TS's strategy changes? If so, maybe we should include some permalinks to the relevant TS source here.

Reply:
Link added.